lib/ModalPresenter.tsx

```diff
@@ -38,16 +97,82 @@ export const showModal = <ContentProps,>(
       </ModalPresenter>
     ),
   );
-  return {
-    dismiss: dismiss,
-  } as ModalHandler;
+  return { dismiss } as ModalHandler;
 };
 
-type Ref = {
-  animatedOut: (completion?: () => void) => void;
+/**
+ * Add a modal to a queue.
+ * 
+ * By default, this method adds the modal to the default queue.
+ * 
+ * [Learn how queue works.](https://github.com/whitespectre/rn-modal-presenter#presenting-modals-in-a-queue)
+ * 
+ * @param Content A component to be presented as a modal on screen. 
+ * This component will be centered horizontally and vertically on screen with
+ * a semitransparent black overlay underneath.
+ * @param contentProps Props for this modal component.
+ * @param options Specify a `ModalOptions` to tweak how this modal will be presented in which queue.
+ * @returns A `Promise` containing a `ModalHandler` when the modal is presented on screen. 
+ * You can use the handler to dismiss the modal imperatively.
+ */
+export const enqueueModal = async <ContentProps,>(
+  Content: (props: ContentProps & ModalContentProps) => JSX.Element,
+  contentProps: ContentProps,
+  options?: ModalOptions,
+) => {
+  const manager = ModalQueuesManager.shared;
+  const _queueName = options?.queueName ?? defaultQueueName;
+  const queue = manager.findQueueByName(_queueName) ?? manager.addQueue(_queueName);
+  return new Promise<ModalHandler>(resolve => {
+    let ref: ModalPresenterRef | null = null;
+    let rootSiblings: RootSiblings | null = null;
+    let element: ModalQueueElement;
+    let onDismissForQueue: (() => void) | null = null;
+
+    const dismiss = (onDismiss?: () => void) => {
+      if (rootSiblings) {
+        if (ref) {
+          // FIXME: set up a flag here to prevent multiple `dismiss` calls during the out animation.
+          ref.animatedOut(() => {
+            rootSiblings?.destroy();
```

Comment:
these two blocks are similar, can we rewrite this as follows?

```
const dismissModal = () => {
  rootSiblings.destroy();
  rootSiblings = null;
  options?.onDismiss?.();
  onDismissForQueue?.();
  onDismiss?.();
}

if (ref) {
  ref.animatedOut(() => {
    ref = null;
    dismissModal();
  });
} else {
  console.warn('Dismissing a modal without animation because ref has been lost.');
  dismissModal();
}
```

Reply:
makes sense. I'll apply your suggestion.